usaspending_api/recipient/v2/views/list_recipients.py

```diff
@@ -70,26 +79,77 @@ def get_recipients(filters={}):
     return results, page_metadata
 
 
+def duns_count_key_function(view_instance, view_method, request, args, kwargs):
+    """
+    A custom cache key is created for this endpoint to ensure that only 'award_type'
+    and 'keyword' fields are used from the request body. Other fields that may come in
+    when this view is called from the RecipientList endpoint, such as 'page', should
+    be ignored.
+    """
+
+    validated_payload = TinyShield(RECIPIENT_MODELS).block(request.data)
```

Comment:
Will this do sanitization of the payload data, or otherwise validate that it conforms to something like strings and numbers. 
If this is becoming part of the key for a Redis cache entry, there is a risk that an injection attack could be used here. This may need some research into:

1. How django-redis or django-cache uses keys (does it has or scrub them before using them?)
2. If not, if you can borrow code from that framework for how it sanitizes the keys first in the default algorithm

Specifically this is a risk for the `keyword` field if I can assume TinyShield is validating that award_type is one of a known set of enum values.

Reply:
Keith, this has been reworked to make use of our caching library. The values coming from the user are hashed before stored in the cache

Reply:
> Will this do sanitization of the payload data, or otherwise validate that it conforms to something like strings and numbers.

Yes, that is one of the primary reasons for using TinyShield